### PR TITLE
fix: github-actions plugin's dev environment properly renders the plugin components

### PR DESCRIPTION
**Pull request by mdb**
This seeks to fix issue #16181 to properly render the plugin component when running `yarn start` in local development.

### Before

Screen Shot 2023-02-05 at 8 04 05 AM

### After

The `/github-actions` route view:

Screen Shot 2023-02-05 at 8 11 00 AM

The `/github-actions/30433642` workflow run route view:

Screen Shot 2023-02-05 at 8 21 20 AM

The `/github-actions/30433642` workflow run route view with the "Build" job expanded:

Screen Shot 2023-02-05 at 8 22 32 AM

The `/github-actions/30433642` workflow run route view with the "Build" job's job log expanded:

Screen Shot 2023-02-05 at 8 24 15 AM

#### :heavy_check_mark: Checklist

- [X] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [X] Screenshots attached (for UI changes)
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/github-actions/src/components/Cards/RecentWorkflowRunsCard.test.tsx [hunk 1]
```diff
@@ -94,15 +94,15 @@ describe('<RecentWorkflowRunsCard />', () => {
     );
 
   it('renders a table with a row for each workflow', async () => {
-    const subject = await renderSubject();
```
Comment (mdb) [inline plugins/github-actions/src/components/Cards/RecentWorkflowRunsCard.test.tsx:97]: As a bonus, I've also removed these unnecessary errant `await`s, as `renderSubject` is not async :)

File: .changeset/neat-elephants-breathe.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-github-actions': patch
+---
+
+`yarn start` now starts a functional dev environment using a mock GitHub API to properly render the `<EntityGithubActionsContent />` component.
```
Comment (mdb) [inline .changeset/neat-elephants-breathe.md:5]: Should this changeset be removed given that the PR does not actually alter any user-facing `github-actions` functionality?

Comment (freben) [reply .changeset/neat-elephants-breathe.md:5]: yep that's correct!

**Review by freben — COMMENTED** [review]
Looks good, but we can remove the changeset indeed.

File: .changeset/neat-elephants-breathe.md (hunk 2 above)
Comment (mdb) [reply .changeset/neat-elephants-breathe.md]: Thanks @freben - I've removed the changeset!

**Review by freben — APPROVED** [review]

**Pull request merged**